anomalib/deploy/inferencers/openvino_inferencer.py

```diff
@@ -132,18 +140,51 @@ def post_process(
                 Defaults to None.
 
         Returns:
-            np.ndarray: Post processed predictions that are ready to be visualized.
+            Dict[str, Any]: Post processed prediction results.
         """
         if meta_data is None:
             meta_data = self.meta_data
 
         predictions = predictions[self.output_blob]
-        anomaly_map = predictions.squeeze()
-        pred_score = anomaly_map.reshape(-1).max()
 
-        anomaly_map, pred_score = self._normalize(anomaly_map, pred_score, meta_data)
+        # Initialize the result variables.
+        anomaly_map: Optional[np.ndarray] = None
+        pred_label: Optional[float] = None
+        pred_mask: Optional[float] = None
 
-        if "image_shape" in meta_data and anomaly_map.shape != meta_data["image_shape"]:
-            anomaly_map = cv2.resize(anomaly_map, meta_data["image_shape"])
-
-        return anomaly_map, float(pred_score)
+        # If predictions returns a single value, this means that the task is
+        # classification, and the value is the classification prediction score.
+        if len(predictions.shape) == 1:
+            task = "classification"
+            pred_score = predictions.item()
+        else:
+            task = "segmentation"
+            anomaly_map = predictions.squeeze()
+            pred_score = anomaly_map.reshape(-1).max()
+
+        # Common practice in anomaly detection is to assign anomalous
+        # label to the prediction if the prediction score is greater
+        # than the image threshold.
+        if "image_threshold" in meta_data:
+            pred_label = pred_score >= meta_data["image_threshold"]
+
+        if task == "segmentation":
+            if "pixel_threshold" in meta_data:
+                pred_mask = (anomaly_map >= meta_data["pixel_threshold"]).astype(np.uint8)
+
+            anomaly_map, pred_score = self._normalize(anomaly_map, pred_score, meta_data)
+
+            if "image_shape" in meta_data and anomaly_map.shape != meta_data["image_shape"]:
+                image_height = meta_data["image_shape"][0]
+                image_width = meta_data["image_shape"][1]
+                anomaly_map = cv2.resize(anomaly_map, (image_width, image_height))
+
+                if pred_mask is not None:
+                    pred_mask = cv2.resize(pred_mask, (image_width, image_height))
+
+        return {
+            "anomaly_map": anomaly_map,
```

Comment:
Would this work for task type of classification?

Reply:
Yeah, it returns none, which ImageResults handles ok.